src/test/java/org/jenkinsci/plugins/gitclient/CredentialsTest.java

```diff
@@ -295,21 +306,88 @@ public static Collection gitRepoUrls() throws MalformedURLException, FileNotFoun
             }
         }
         Collections.shuffle(repos); // randomize test order
-        int toIndex = Math.min(repos.size(), TEST_ALL_CREDENTIALS ? 120 : 6); // Don't run more than 120 variations of test - about 3 minutes
-        return repos.subList(0, toIndex);
+        // If we're not testing all credentials, take 6 or less
+        return TEST_ALL_CREDENTIALS ? repos : repos.subList(0, Math.min(repos.size(), 6));
+    }
+
+    private void doFetch(String source) throws InterruptedException, URISyntaxException {
+        /* Save some bandwidth with shallow clone for CliGit, not yet available for JGit */
+        URIish sourceURI = new URIish(source);
+        List<RefSpec> refSpecs = new ArrayList<>();
+        refSpecs.add(new RefSpec("+refs/heads/master:refs/remotes/origin/master"));
+        FetchCommand cmd = git.fetch_().from(sourceURI, refSpecs).tags(false);
+        if (gitImpl.equals("git")) {
+            // Reduce network transfer by using shallow clone
+            // JGit does not support shallow clone
+            cmd.shallow(true).depth(1);
+        }
+        cmd.execute();
     }
 
-    private void addCredential(String username, String password, File privateKey) throws IOException {
-        if (random.nextBoolean()) {
+    private String listDir(File dir) {
+        File[] files = repo.listFiles();
+        StringBuilder fileList = new StringBuilder();
```

Comment:
Could be simpler to use [StringJoiner](https://docs.oracle.com/javase/8/docs/api/java/util/StringJoiner.html).

Reply:
Yes, since this branch uses Java 8, that would be better.

The changes in #337 can't use StringJoiner because that version can only use Java 7 language features.